go.mod

```diff
@@ -107,6 +107,7 @@ require (
 	github.com/julienschmidt/httprouter v1.3.0 // replaced
 	github.com/keys-pub/go-libfido2 v1.5.3-0.20220306005615-8ab03fb1ec27 // replaced
 	github.com/kyroy/kdtree v0.0.0-20200419114247-70830f883f1d
+	github.com/lox/bintest v2.0.0+incompatible
```

Comment:
Looks like this is an old go mod incompatible version. Can you try using [github.com/buildkite/bintest/v3](https://pkg.go.dev/github.com/buildkite/bintest/v3) instead?

Reply:
I'm not sure why I didn't pull the latest version.
Well, `v3` doesn't work when you move the binary out of place because it matches against the full binary path.
When the mock binary is created it stores its name, eg `/tmp/bintest.../teleport`
When we compress, serve, download, extract and then try to execute the binary, bintest doesn't recognize that binary because it uses a full match on the binary path. The tarball is downloaded to the result of `t.TempDir()` which is different from `/tmp/bintest.../teleport`.

To circumvent this issue, I changed things a little bit but the end result is the same and we are able to test the full flow.
Automatic tests pass, and I did a manual test as well :+1: 
I'll merge the PR, but feel free to comment/DM and we can change the testing approach.

Thank you for your review.